lib/Interpreter/Transaction.cpp

```diff
@@ -62,12 +62,22 @@ namespace cling {
   }
 
   NamedDecl* Transaction::containsNamedDecl(llvm::StringRef name) const {
-    for (auto I = decls_begin(), E = decls_end(); I < E; ++I)
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
-        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI))
+        if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
           if (name.equals(ND->getNameAsString()))
             return ND;
+        }
+        else if (LinkageSpecDecl* LSD = dyn_cast<LinkageSpecDecl>(DI)) {
```

Comment:
There is only one (mis?)use and I was wondering whether we should fix the use or keep this interface. Right now it is a misnomer, because it checks if the transaction contains a decl, but also returns it (which is a sort-of lookup).

It seems reasonable to me to rename this into `bool cling::Transaction::containsDecl(Decl*);` and add a Sema lookup in our `MetaSema`.